### ISSUE TITLE
problems with dr2xml

### ISSUE THREAD

**Issue by etiennesky** [goord/ece2cmor3#124]
Hi,
I am not sure if this is the proper place to report this, but I am asking here first since you guys know your stuff! Apologies if I whould post this elsewhere.

I was following the Dr2xml issues page and followed all the steps to install the libraries and repositories.

I got errors running both ece-dr2xml.py and ece-create-ping-files.py , maybe the master branch is unstable?

```
etourign@bscearth346:~/cmorize/dr2xml>  cd ${HOME}/cmorize/dr2xml/dr2xml-repository/
etourign@bscearth346:~/cmorize/dr2xml/dr2xml-repository>   python ece-dr2xml.py
VERSIONS:
---------
* dr2xml version:  pre-0.28
CMIP6 conventions version: v6.2.4
* CMIP6 Data Request version:  01.00.21

LAB AND MODEL SETTINGS:
------------------------
*  listof_home_vars = ./input_labs/cerfacs/home_vars/listof_primavera_extra_vars.txt
*  tierMax = 3
*  grids = {'HR': {'nemo': ['gn', '', '25 km', 'native ocean tri-polar grid with 1.47 M ocean cells'], 'arpsfx': ['gr', 'completeHR', '50 km', 'data regridded to a 359 gaussian grid (180x360 latlon) from a native atmosphere T359l reduced gaussian grid']}, 'LR': {'nemo': ['gn', '', '100 km', 'native ocean tri-polar grid with 105 k ocean cells'], 'arpsfx': ['gr', 'complete', '250 km', 'data regridded to a T127 gaussian grid (128x256 latlon) from a native atmosphere T127l reduced gaussian grid']}}
*  path_to_parse = ./input_labs/cerfacs/to_parse/for_arpege_grid_dr2xml_LR_STMARTIN
*  excluded_vars = ['clcalipso', 'clcalipso2', 'clhcalipso', 'clcalipsoice', 'clcalipsoliq', 'cllcalipso', 'clmcalipso', 'cltcalipso', 'cfadLidarsr532', 'cfadDbze94', 'cfadLidarsr532', 'clisccp', 'cltisccp', 'climodis', 'clwmodis', 'jpdftaureicemodis', 'cltmodis', 'clmisr', 'parasolRefl']
*  model_timestep = {'nemo': 900.0, 'arpsfx': 900.0}
*  realms_per_context = {'nemo': ['seaIce', 'ocean', 'ocean seaIce', 'ocnBgchem', 'seaIce ocean'], 'arpsfx': ['atmos', 'atmos atmosChem', 'aerosol', 'atmos land', 'land', 'landIce land', 'aerosol land', 'land landIce', 'landIce']}
*  max_file_size_in_floats = 500000000.0
*  path_extra_tables = ./input_labs/cerfacs/extra_Tables
*  ping_variables_prefix = CMIP6_
*  comments = {}
*  grid_choice = {'CNRM-ESM2-1': 'LR', 'CNRM-ESM2-1-HR': 'HR', 'CNRM-CM6-1': 'LR', 'CNRM-CM6-1-HR': 'HR'}
*  orphan_variables = {'nemo': ['dummy_variable_for_illustration_purpose'], 'arpsfx': []}
*  vars_OK = {}
*  max_priority = 2
*  info_url = http://www.umr-cnrm.fr/cmip6/
*  institution_id = CNRM-CERFACS
*  excluded_spshapes = ['XYA-na', 'XYG-na', 'S-na', 'S-AH', 'S-A', 'na-A', 'Y-P39']
*  mips = HighResMIP
*  use_union_zoom = False
*  source_types = {'CNRM-ESM2-1': 'ESM', 'CNRM-ESM2-1-HR': 'ESM', 'CNRM-CM6-1': 'AOGCM', 'CNRM-CM6-1-HR': 'AOGCM'}
*  sizes = [105704, 75, 32768, 91, 30, 14, 128]
*  mips_all = set(['C4MIP', 'CORDEX', 'ISMIP6', 'GMMIP', 'LUMIP', 'PMIP', 'RFMIP', 'DAMIP', 'AerChemMIP', 'SIMIP', 'FAFMIP', 'CFMIP', 'ScenarioMIP', 'LS3MIP', 'OMIP', 'GeoMIP'])
*  contact = contact.cmip6@cerfacs.fr
*  output_path = @IOXDIR@/
*  grid_policy = adhoc
*  source_id = CNRM-CM6-1

SIMULATION SETTINGS:
---------------------
*  activity_id = HighResMIP
*  parent_source_id = CNRM-CM5.1
*  variant_info = none
*  parent_experiment_id = no parent
*  parent_activity_id = CMIP
*  source_type = AGCM
*  physics_index = 1
*  sub_experiment_id = none
*  branch_time_in_parent = 365.0D0
*  project = CMIP6
*  sub_experiment = none
*  branch_method = standard
*  initialization_index = 1
*  realization_index = 1
*  comments = {}
*  forcing_index = 1
*  branch_time_in_child = 0.0D0
*  parent_mip_era = CMIP5
*  experiment_id = Forced-Atmos-Land
*  history = none
VERSIONS:
---------

LAB AND MODEL SETTINGS:
------------------------
*  listof_home_vars = ./input_labs/cerfacs/home_vars/listof_primavera_extra_vars.txt
*  tierMax = 3
*  grids = {'HR': {'nemo': ['gn', '', '25 km', 'native ocean tri-polar grid with 1.47 M ocean cells'], 'arpsfx': ['gr', 'completeHR', '50 km', 'data regridded to a 359 gaussian grid (180x360 latlon) from a native atmosphere T359l reduced gaussian grid']}, 'LR': {'nemo': ['gn', '', '100 km', 'native ocean tri-polar grid with 105 k ocean cells'], 'arpsfx': ['gr', 'complete', '250 km', 'data regridded to a T127 gaussian grid (128x256 latlon) from a native atmosphere T127l reduced gaussian grid']}}
*  path_to_parse = ./input_labs/cerfacs/to_parse/for_arpege_grid_dr2xml_LR_STMARTIN
*  excluded_vars = ['clcalipso', 'clcalipso2', 'clhcalipso', 'clcalipsoice', 'clcalipsoliq', 'cllcalipso', 'clmcalipso', 'cltcalipso', 'cfadLidarsr532', 'cfadDbze94', 'cfadLidarsr532', 'clisccp', 'cltisccp', 'climodis', 'clwmodis', 'jpdftaureicemodis', 'cltmodis', 'clmisr', 'parasolRefl']
*  model_timestep = {'nemo': 900.0, 'arpsfx': 900.0}
*  realms_per_context = {'nemo': ['seaIce', 'ocean', 'ocean seaIce', 'ocnBgchem', 'seaIce ocean'], 'arpsfx': ['atmos', 'atmos atmosChem', 'aerosol', 'atmos land', 'land', 'landIce land', 'aerosol land', 'land landIce', 'landIce']}
*  max_file_size_in_floats = 500000000.0
*  path_extra_tables = ./input_labs/cerfacs/extra_Tables
*  ping_variables_prefix = CMIP6_
*  comments = {}
*  grid_choice = {'CNRM-ESM2-1': 'LR', 'CNRM-ESM2-1-HR': 'HR', 'CNRM-CM6-1': 'LR', 'CNRM-CM6-1-HR': 'HR'}
*  orphan_variables = {'nemo': ['dummy_variable_for_illustration_purpose'], 'arpsfx': []}
*  vars_OK = {}
*  max_priority = 2
*  info_url = http://www.umr-cnrm.fr/cmip6/
*  institution_id = CNRM-CERFACS
*  excluded_spshapes = ['XYA-na', 'XYG-na', 'S-na', 'S-AH', 'S-A', 'na-A', 'Y-P39']
*  mips = HighResMIP
*  use_union_zoom = False
*  source_types = {'CNRM-ESM2-1': 'ESM', 'CNRM-ESM2-1-HR': 'ESM', 'CNRM-CM6-1': 'AOGCM', 'CNRM-CM6-1-HR': 'AOGCM'}
*  sizes = [105704, 75, 32768, 91, 30, 14, 128]
*  mips_all = set(['C4MIP', 'CORDEX', 'ISMIP6', 'GMMIP', 'LUMIP', 'PMIP', 'RFMIP', 'DAMIP', 'AerChemMIP', 'SIMIP', 'FAFMIP', 'CFMIP', 'ScenarioMIP', 'LS3MIP', 'OMIP', 'GeoMIP'])
*  contact = contact.cmip6@cerfacs.fr
*  output_path = @IOXDIR@/
*  grid_policy = adhoc
*  source_id = CNRM-CM6-1

SIMULATION SETTINGS:
---------------------
*  activity_id = HighResMIP
*  parent_source_id = CNRM-CM5.1
*  variant_info = none
*  parent_experiment_id = no parent
*  parent_activity_id = CMIP
*  source_type = AGCM
*  physics_index = 1
*  sub_experiment_id = none
*  branch_time_in_parent = 365.0D0
*  project = CMIP6
*  sub_experiment = none
*  branch_method = standard
*  initialization_index = 1
*  realization_index = 1
*  comments = {}
*  forcing_index = 1
*  branch_time_in_child = 0.0D0
*  parent_mip_era = CMIP5
*  experiment_id = Forced-Atmos-Land
*  history = none
Traceback (most recent call last):
  File "ece-dr2xml.py", line 483, in <module>
    cvs_path=my_cvspath,dummies='skip', dirname=my_dir)
TypeError: generate_file_defs() got an unexpected keyword argument 'pingfile'
```

```
etourign@bscearth346:~/cmorize/dr2xml/dr2xml-repository> cp ${HOME}/cmorize/dr2xml/dr2xml-repository/doc/create_ping_files.py ${HOME}/cmorize/dr2xml/dr2xml-repository/ece-create-ping-files.py
etourign@bscearth346:~/cmorize/dr2xml/dr2xml-repository>   cd ${HOME}/cmorize/dr2xml/dr2xml-repository/
etourign@bscearth346:~/cmorize/dr2xml/dr2xml-repository>   python ece-create-ping-files.py
* dr2xml version:  pre-0.28
CMIP6 conventions version: v6.2.4
* CMIP6 Data Request version:  01.00.21
Traceback (most recent call last):
  File "ece-create-ping-files.py", line 128, in <module>
    svars=gather_AllSimpleVars(settings,printout=True)
TypeError: gather_AllSimpleVars() takes at least 2 arguments (2 given)
```

**Comment by treerink** [comment]
Hi Etienne,

Running the dr2xml is not out of the box at all. We had a lot of trouble to get it at least running (still probably not correct way of running yet). But many of the previous hacks which we had to apply to get it running are now solved at the dr2xml repro and since yesterday the master branch is the one to use now (which makes it more straightforward). Yesterday a few of our changes have been merged in to the dr2xml repro. However, we need still to apply two severe hacks to get it running and a totally rewritten ece_dr2xml.py and ece_create_ping_files.py. I have been busy with that today as well. For the major hacks I will contact the developers, and see if it is easy to fix. In order to use ece_dr2xml.py within EC-EARTH3 we have also the NEMO update issue which is directly related to this.

If you like I can share the code status I have now, but as this is indeed off topic here, I suggest we continue by mail.

Thomas

Closing this issue as it is off topic here.
